### PR TITLE
Change #,# delimiter in multi-select case search fields

### PR DESCRIPTION
**Pull request by orangejenny**
## Product Description
https://dimagi-dev.atlassian.net/browse/USH-2382

## Technical Description
Interdependent with https://github.com/dimagi/commcare-core/pull/1149

### Safety story
There's reasonable test coverage of this logic.

### Automated test coverage
In PR

### QA Plan
Requesting QA

### Special deploy instructions
This needs to be deployed at the same time as  and https://github.com/dimagi/commcare-hq/pull/32139

### Rollback instructions
If this is rolled back, https://github.com/dimagi/commcare-hq/pull/32139 also needs to be rolled back.

This isn't ideal, but I'm not terribly worried about this set of changes needing to be rolled back.

### Review

- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change.

### CODE REVIEW

**Review by snopoke — APPROVED** [review]

File: src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java [hunk 1]
```diff
@@ -63,8 +64,7 @@ public QueryResponseBean(QueryScreen queryScreen, SessionWrapper session) {
                     }
                 }
                 if (indicesForSelectedChoices.size() > 0) {
-                    currentAnswer = String.join(RemoteQuerySessionManager.ANSWER_DELIMITER,
-                            indicesForSelectedChoices);
+                    currentAnswer = DataUtil.joinWithSpaces(indicesForSelectedChoices.toArray(new String[0]));
```
Comment (shubham1g5) [inline src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java:67]: we should initialise the array with the exact size like - 
```suggestion
                    currentAnswer = DataUtil.joinWithSpaces(indicesForSelectedChoices.toArray(new String[indicesForSelectedChoices.size()]));
```

**Review by shubham1g5 — CHANGES_REQUESTED** [review]

File: src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java (hunk 1 above)
Comment (orangejenny) [reply src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java]: b744e383

**Review by shubham1g5 — APPROVED** [review]

File: src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java [hunk 2]
```diff
@@ -63,8 +64,8 @@ public QueryResponseBean(QueryScreen queryScreen, SessionWrapper session) {
                     }
                 }
                 if (indicesForSelectedChoices.size() > 0) {
-                    currentAnswer = String.join(RemoteQuerySessionManager.ANSWER_DELIMITER,
```
Comment (shubham1g5) [inline src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java:66]: We would need this to be backward compatible as well to start with to remove the dependency between HQ and Formplayer deploy.

**Review by shubham1g5 — CHANGES_REQUESTED** [review]

**Comment by orangejenny** [comment]
Closing, see discussion on ticket.

**Pull request closed without merging**